File: 05.Stack/Data structure Stack.java
import java.util.*;

public class data structure Stack {
    public static void main(String[] args) {
        
        // Stack declaration and creation
        Deque<Integer> stack = new ArrayDeque<Integer>();
        
        stack.push(1); // 1 added
        stack.push(2); // 2 added
        stack.push(3); // 3 added
        System.out.println(stack.peek()); // Top data verification : 3
        
        stack.push(4); // 4 added
        stack.push(5); // 5 added
        System.out.println(stack.peek()); // Top data verification : 5

        System.out.println(stack.pop()); // Top data verification and removal :5
        System.out.println(stack.peek()); // Top data verification : 4
        
        // Removes all data in the Stack and print
        while(!stack.isEmpty())
        {
            System.out.println(stack.pop());
        }
    }
}

File: 05.Stack/Data Structure Stack.cpp
#include <iostream>
// stack header include
#include <stack>
using namespace std;

int main()
{
    // Stack declaration and creation
    stack<int> stk;
    
    stk.push(1); // 1 added
    stk.push(2); // 2 added
    stk.push(3); // 3 added
    cout << stk.top() << "\n"; // Top data verification : 3
    
    stk.push(4); // 4 added
    stk.push(5); // 5 added
    cout << stk.top() << "\n"; // Top data verification : 5

    stk.pop(); // Top data removal : 5 remove
    cout << stk.top() << "\n"; // Top data verification : 4
    
    // Removes all data in the Stack and print
    while(!stk.empty())
    {
        cout << stk.top() << "\n";
        stk.pop();
    }    
}
